pkg/executor/executortype/poolmgr/gpm.go

```diff
@@ -682,3 +693,90 @@ func (gpm *GenericPoolManager) idleObjectReaper() {
 		}
 	}
 }
+
+// WebsocketEventChecker checks if the pod has emitted a websocket event
+func (gpm *GenericPoolManager) WebsocketEventChecker(kubeClient *kubernetes.Clientset) {
+
+	informer := k8scache.NewSharedInformer(
+		&k8scache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				options.FieldSelector = "involvedObject.kind=Pod,type=Normal,reason=Websocket"
+				return kubeClient.CoreV1().Events(apiv1.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				options.FieldSelector = "involvedObject.kind=Pod,type=Normal,reason=Websocket"
+				return kubeClient.CoreV1().Events(apiv1.NamespaceAll).Watch(options)
+			},
+		},
+		&apiv1.Event{},
+		0,
+	)
+
+	stopper := make(chan struct{})
+	defer close(stopper)
+	informer.AddEventHandler(k8scache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			mObj := obj.(metav1.Object)
+			gpm.logger.Info("Websocket event detected for pod",
+				zap.String("Pod name", mObj.GetName()))
+
+			podName := strings.SplitAfter(mObj.GetName(), ".")
+			if fsvc, ok := gpm.fsCache.PodToFsvc[strings.TrimSuffix(podName[0], ".")]; ok {
+				gpm.fsCache.WebsocketFsvc[fsvc.Name] = true
+			}
+		},
+	})
+	informer.Run(stopper)
+
+}
+
+// InactiveEventChecker checks if the pod has emitted an inactive event
+func (gpm *GenericPoolManager) InactiveEventChecker(kubeClient *kubernetes.Clientset) {
+
+	informer := k8scache.NewSharedInformer(
+		&k8scache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				options.FieldSelector = "involvedObject.kind=Pod,type=Normal,reason=Inactive"
```

Comment:
`Inactive` is a too generic an event name and going to be used by others so easy to confuse, I would suggest we come up with a properly qualified names for events so that we can clearly identify. For ex. something like FISSION_WS_ACTIVATED/TERMINATED or something of that sort.

Reply:
For initiating, I am thinking `InitiateLongLivedConnection`  . I am not sure if we should use the term `terminate` . In the environment, we're actually checking if there are any active connections and when we find that there are no active connections, we send out an event so technically there are no connections to terminate. How about just calling it what is: `NoActiveConnections` ?

Reply:
`NoActiveConnections` Looks good. Also check related comment earlier!